### ISSUE TITLE
roadmap: Search within Jan app

### ISSUE THREAD

**Issue by imtuyethan** [menloresearch/jan#3648]
# Related
- [ ] https://github.com/janhq/jan/issues/2930

# Problem Statement

To make it easier & faster to access information in Jan App.

# Feature Idea
Search Functionality: To work at 3 levels: 

### 1. Global, across the entire app
Note: Likely implement after #4255

**Unified search across:**
- Chat threads & messages
- Settings & preferences
- Models & assistants
- Extensions & commands
- Files & attachments 

### 2. Within thread folder
- Quick navigation to specific threads or messages
- Shows results grouped by threads
- [Out-of-scope] Search files & images
- [Out-of-scope] Filters: date range, message type (user/assistant)
- Access: Search bar in thread list sidebar
- Results: Highlight matching threads

Figma link: https://www.figma.com/design/DYfpMhf8qiSReKvYooBgDV/Jan-App-(3rd-version)?node-id=9723-54549&m=dev

<img width="702" alt="Screenshot 2024-12-26 at 9 08 38 AM" src="https://github.com/user-attachments/assets/6660b6ac-01ec-42cd-8111-001a86d91f23" />

### 3. Within the open thread or within tree-branch/group 
- Scope: Active conversation
- Access: Search bar in thread view or through shortcut (`CMD + F` or `CTR + F`)
- Results: Highlight matching messages
- Jump between results with up/down controls
- Include message content and metadata in the search

Figma: https://www.figma.com/design/DYfpMhf8qiSReKvYooBgDV/Jan-App-(3rd-version)?node-id=9720-30924&m=dev

<img width="641" alt="Screenshot 2024-12-26 at 8 28 54 AM" src="https://github.com/user-attachments/assets/645ac027-29e9-42d5-ac4b-95bbf2392483" />

**Comment by imtuyethan** [comment]
I did wireframes months ago where users can have search bars in:
- Top nav (for global search)
- On top of threads list (to search for relevant thread + messages faster)

**Comment by imtuyethan** [comment]
<img width="802" alt="Screenshot 2024-09-20 at 4 19 28 PM" src="https://github.com/user-attachments/assets/3476124e-6403-4d4a-8886-facd155d8a5d">

**Comment by imtuyethan** [comment]
Related: https://github.com/janhq/jan/issues/3772

**Comment by dan-homebrew** [comment]
@imtuyethan I am reclassifying this as a Roadmap type, as Search is non-trivial and requires a lot of engineering effort.

**Comment by louis-menlo** [comment]
Scope change: 
- As we are simplifying the UX, thread search is supported in the new FE so that this could be closed, see #4884
